File: 1603-RunningSumOf1dArray/1603-RunningSumOf1dArray.java
// Last updated: 8/13/2025, 8:17:27 PM
class Solution {
    public int[] runningSum(int[] nums) {
        int n = nums.length;
        int[]p = new int[n];
        p[0] = nums[0];
        for(int i=1;i<n;i++){
            p[i] = p[i-1] + nums[i];
        }
        return p;
    }
}

File: 1603-RunningSumOf1dArray/1603-RunningSumOf1dArray.py
# Last updated: 8/13/2025, 8:17:21 PM
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        result = [None]*len(nums)
        result[0] = nums[0]
        for i in range(1,len(nums)) : 
            result[i] = result[i-1] + nums[i]
        return result